### ISSUE TITLE
Unhandled Exception Crash when restoring backup from GCP

### ISSUE THREAD

**Issue by Auberixli1** [duplicati/duplicati#3606]
- [x] I have searched open and closed issues for duplicates.

----------------------------------------

## Environment info

- **Duplicati version**: 2.0.4.9
- **Operating system**: Fedora 29
- **Backend**: GCP

## Description

Throws Unhanded exception, System.TypeInitializationException, when restoring from a backup on GCP.

## Steps to reproduce

1. Go though the steps to restore an encrypted backup from GCP.

- **Actual result**:
Throws the previously mention exception.
- **Expected result**:
To go to the next step of the restoring process.

## Debug log

```Unhandled Exception:
System.TypeInitializationException: The type initializer for 'System.Console' threw an exception. ---> System.TypeInitializationException: The type initializer for 'System.ConsoleDriver' threw an exception. ---> System.Exception: Magic number is wrong: 542
  at System.TermInfoReader.ReadHeader (System.Byte[] buffer, System.Int32& position) [0x0002b] in <373b6e083d6e45e498c9082a8eebd27f>:0 
  at System.TermInfoReader..ctor (System.String term, System.String filename) [0x00065] in <373b6e083d6e45e498c9082a8eebd27f>:0 
  at System.TermInfoDriver..ctor (System.String term) [0x00058] in <373b6e083d6e45e498c9082a8eebd27f>:0 
  at System.ConsoleDriver.CreateTermInfoDriver (System.String term) [0x00000] in <373b6e083d6e45e498c9082a8eebd27f>:0 
  at System.ConsoleDriver..cctor () [0x00062] in <373b6e083d6e45e498c9082a8eebd27f>:0 
   --- End of inner exception stack trace ---
  at System.Console.SetupStreams (System.Text.Encoding inputEncoding, System.Text.Encoding outputEncoding) [0x0000a] in <373b6e083d6e45e498c9082a8eebd27f>:0 
  at System.Console..cctor () [0x000a8] in <373b6e083d6e45e498c9082a8eebd27f>:0 
   --- End of inner exception stack trace ---
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunMethod (System.Reflection.MethodInfo method, System.String[] args) [0x00097] in <3931343baab74f90ae3b1f2dfa9e1f83>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager+<>c__DisplayClass58_1.<RunFromMostRecentSpawn>b__0 () [0x00000] in <3931343baab74f90ae3b1f2dfa9e1f83>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager.WrapWithUpdater (Duplicati.Library.AutoUpdater.AutoUpdateStrategy defaultstrategy, System.Action wrappedFunction) [0x00157] in <3931343baab74f90ae3b1f2dfa9e1f83>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecentSpawn (System.Reflection.MethodInfo method, System.String[] cmdargs, Duplicati.Library.AutoUpdater.AutoUpdateStrategy defaultstrategy) [0x0004b] in <3931343baab74f90ae3b1f2dfa9e1f83>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent (System.Reflection.MethodInfo method, System.String[] cmdargs, Duplicati.Library.AutoUpdater.AutoUpdateStrategy defaultstrategy) [0x0001b] in <3931343baab74f90ae3b1f2dfa9e1f83>:0 
  at Duplicati.GUI.TrayIcon.Program.Main (System.String[] args) [0x0001c] in <1690e3f004c448538c253fe338464d73>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: The type initializer for 'System.Console' threw an exception. ---> System.TypeInitializationException: The type initializer for 'System.ConsoleDriver' threw an exception. ---> System.Exception: Magic number is wrong: 542
  at System.TermInfoReader.ReadHeader (System.Byte[] buffer, System.Int32& position) [0x0002b] in <373b6e083d6e45e498c9082a8eebd27f>:0 
  at System.TermInfoReader..ctor (System.String term, System.String filename) [0x00065] in <373b6e083d6e45e498c9082a8eebd27f>:0 
  at System.TermInfoDriver..ctor (System.String term) [0x00058] in <373b6e083d6e45e498c9082a8eebd27f>:0 
  at System.ConsoleDriver.CreateTermInfoDriver (System.String term) [0x00000] in <373b6e083d6e45e498c9082a8eebd27f>:0 
  at System.ConsoleDriver..cctor () [0x00062] in <373b6e083d6e45e498c9082a8eebd27f>:0 
   --- End of inner exception stack trace ---
  at System.Console.SetupStreams (System.Text.Encoding inputEncoding, System.Text.Encoding outputEncoding) [0x0000a] in <373b6e083d6e45e498c9082a8eebd27f>:0 
  at System.Console..cctor () [0x000a8] in <373b6e083d6e45e498c9082a8eebd27f>:0 
   --- End of inner exception stack trace ---
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunMethod (System.Reflection.MethodInfo method, System.String[] args) [0x00097] in <3931343baab74f90ae3b1f2dfa9e1f83>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager+<>c__DisplayClass58_1.<RunFromMostRecentSpawn>b__0 () [0x00000] in <3931343baab74f90ae3b1f2dfa9e1f83>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager.WrapWithUpdater (Duplicati.Library.AutoUpdater.AutoUpdateStrategy defaultstrategy, System.Action wrappedFunction) [0x00157] in <3931343baab74f90ae3b1f2dfa9e1f83>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecentSpawn (System.Reflection.MethodInfo method, System.String[] cmdargs, Duplicati.Library.AutoUpdater.AutoUpdateStrategy defaultstrategy) [0x0004b] in <3931343baab74f90ae3b1f2dfa9e1f83>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent (System.Reflection.MethodInfo method, System.String[] cmdargs, Duplicati.Library.AutoUpdater.AutoUpdateStrategy defaultstrategy) [0x0001b] in <3931343baab74f90ae3b1f2dfa9e1f83>:0 
  at Duplicati.GUI.TrayIcon.Program.Main (System.String[] args) [0x0001c] in <1690e3f004c448538c253fe338464d73>:0```

**Comment by warwickmm** [comment]
I am unable to reproduce this.  What version of mono are you running?  It looks like there was a related bug in mono that was fixed starting with version 5.10.1.45.

**Comment by Auberixli1** [comment]
Sorry, forgot to close this the other day. Fixed with updating mono. However it might be worth updating the dependency list to require the version you have mentioned.
